tests/resolvers/Mutation/updatePost.spec.ts

```diff
@@ -0,0 +1,125 @@
+import "dotenv/config";
+import { Types } from "mongoose";
+import { Post } from "../../../src/models";
+import { MutationUpdatePostArgs } from "../../../src/types/generatedGraphQLTypes";
+import { connect, disconnect } from "../../../src/db";
+import { updatePost as updatePostResolver } from "../../../src/resolvers/Mutation/updatePost";
+import {
+  POST_NOT_FOUND_MESSAGE,
+  USER_NOT_AUTHORIZED_MESSAGE,
+  USER_NOT_FOUND_MESSAGE,
+} from "../../../src/constants";
+import { beforeAll, afterAll, describe, it, expect, vi } from "vitest";
+import { testUserType } from "../../helpers/userAndOrg";
+import { createTestPost, testPostType } from "../../helpers/posts";
+
+let testUser: testUserType;
+let testPost: testPostType;
+
+beforeAll(async () => {
+  await connect();
+  const temp = await createTestPost();
+  testUser = temp[0];
+  testPost = temp[2];
+  const { requestContext } = await import("../../../src/libraries");
+  vi.spyOn(requestContext, "translate").mockImplementation(
+    (message) => message
+  );
+});
+afterAll(async () => {
+  await disconnect();
+});
+
+describe("resolvers -> Mutation -> updatePost", () => {
+  it(`throws NotFoundError if no user exists with _id === context.userId`, async () => {
+    try {
+      const args: MutationUpdatePostArgs = {
+        id: "",
+      };
+
+      const context = {
+        userId: Types.ObjectId().toString(),
+      };
+
+      await updatePostResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(error.message).toEqual(USER_NOT_FOUND_MESSAGE);
+    }
+  });
+
+  it(`throws NotFoundError if no post exists with _id === args.id`, async () => {
+    try {
+      const args: MutationUpdatePostArgs = {
+        id: Types.ObjectId().toString(),
+      };
+
+      const context = {
+        userId: testUser!._id,
+      };
+
+      await updatePostResolver?.({}, args, context);
+    } catch (error: any) {
+      expect(error.message).toEqual(POST_NOT_FOUND_MESSAGE);
+    }
+  });
+
+  it(`throws UnauthorizedError as current user with _id === context.userId is an admin of post with _id === args.id`, async () => {
```

Comment:
This test is useless because catch block never executes. Because testUser is already a creator of of post. Remove this.

Reply:
Done.